automation/tests/authentication.spec.ts

```diff
@@ -1,17 +1,150 @@
+import { registrationData} from '../test-data/authentication_page_data.js';
+import { randomString } from '../utils/services/creatingData.ts';
+import { randomEmail } from '../utils/services/creatingData.ts';
+
+let authenticationPage: AuthenticationPage;
+
+
 
 test.beforeEach(async({page})=>{
-  await page.goto('http://localhost:5173/');
+  await page.goto('/');
+   authenticationPage = new AuthenticationPage(page);
+});
+
+test.describe('@VB-31 - Navigation to/from login page, and navigation to registration form', ()=> {
+  test('@Smoke - Login Button redirects to login page', async ({page})=> {
+    await expect(authenticationPage.pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+    await authenticationPage.loginButton.click();
+    await expect(authenticationPage.loginPageTitle).toHaveText('Login To Your Account');
+  });
+    test('TryOutNow Button redirects to login page', async ({page})=> {
+      await authenticationPage.tryOutButton.click();
+      await expect(authenticationPage.loginPageTitle).toHaveText('Login To Your Account');
+  });
+    test('@Smoke - Exit Button redirects to guest page', async ({page})=> {
+      await authenticationPage.loginButton.click();
+      await authenticationPage.exitButton.click();
+      await expect(authenticationPage.pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+  });
+  test('@Smoke - Verify link \Register by clicking here\ redirects to Registration Form', async ({page})=> {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.linkToRegistration.click();
+    await expect(authenticationPage.registerFormTitle).toHaveText('Register New Account');
+  });
+  test('Exit Button from registration form redirects to guest page', async ({page})=> {
+    await authenticationPage.navigateToRegistrationForm();
+    await authenticationPage.exitButton.click();
+      await expect(authenticationPage.pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+  })
 })
 
-test('Guest page loaded', async ({ page }) => {
-  const pageTitle = page.getByText('Elevate Your Mood, One Track at a Time');
-  // const pageTitle2 = page.locator('.description h1');
-  await expect(pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+test.describe('@VB-10 - Login. Input validation popup messages', () => {
+   test('@Smoke - Successful login', async ({page}) => {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.fillLoginData(process.env.EMAIL!, process.env.PASSWORD!);
+    await authenticationPage.accountLoginButton.click();
+    await expect(page.getByAltText("An icon of a person's profile view")).toBeVisible();
+  });
+  test('Missing password login - valid email, no password', async ({page})=> {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.fillLoginData(process.env.EMAIL!, loginCredentials.PASSWORD_MISSING);
+    await authenticationPage.accountLoginButton.click();
```

Comment:
Lines 57-59 should be a separate function in the page level as it is repeated at least 4 times here. You should create a function called something like `loginWithInvalidOrMissingCredetials(message)` and you would pass a different message in each test.

loginWithInvalidOrMissingCredetials(message) {
    await authenticationPage.accountLoginButton.click();
    await expect(authenticationPage.popupMessage).toBeVisible();
    await expect(authenticationPage.popupMessage).toHaveText(message);
}
call in the test: 
    await authenticationPage.loginWithInvalidOrMissingCredetials("Wrong login data! Check your credentials");

Reply:
I wrote the function as advised. I named it clickLoginAndReceivePopupMessage(). But now in POM we have expect. Is that ok?